app/controllers/api/pods_controller.rb

```
@@ -209,6 +209,43 @@ class PodsController < APIController
         end
       end
 
+      delete '/:name/:version', :requires_owner => true do
+        pod = Pod.find_by_name_and_owner(params[:name], @owner) do
+          json_error(403, 'You are not allowed to delete this pod.')
+        end
+        unless pod
+          json_error(404, 'No pod found with the specified name.')
+        end
+
+        version = pod.versions_dataset.where(:name => params[:version]).first
```

should raise a friendly error if the version is already deleted